src/context/StoreContext.js

```diff
@@ -56,23 +58,42 @@ export default function StoreContextProvider(props) {
     null
   )
 
+  const onBibleRefUIChange = (bId, ch, v) => {
```

Comment:
This is a large local function def. Could we refactor this outside of this function?

Reply:
I don't see a way to do that at the moment. This is done like this, in order to hide the caching of a state - and not cause any re-rendering.
If there is a way to refactor this, then it would to me seem to be a rather large refactor...

Reply:
let's talk about it when you are back